src/meili/map_matching.cc

```diff
@@ -172,8 +158,8 @@ MapMatching::TransitionCost(const StateId& lhs, const StateId& rhs) const
   const auto& right = state(rhs);
   const auto& left_measurement = measurement(left);
   const auto& right_measurement = measurement(right);
-  const auto gc_dist = GreatCircleDistance(left_measurement, right_measurement);
-  const auto clk_dist = ClockDistance(left_measurement, right_measurement);
+  const auto gc_dist = left_measurement.lnglat().Distance(right_measurement.lnglat());
+  const auto clk_dist = right_measurement.epoch_time() - leave_times_[lhs.time()];
```

Comment:
It seems that we haven't consider the case that `right_measurement.epoch_time() == -1` or `leave_times_[lhs.time()] == -1`. Won't it lead to large clk_dist?

Reply:
i was noticing that myself and thought that i handled that somewhere but its slipped my mind. ill go back and look at the original pr that added support for the `time_factor`

Reply:
ok ive added a saftey check, the only way this problem could happen is if someone gives some measurements with times and some without or gives invalid times as input but anyway it should be safe now